canopen_motor_node/CMakeLists.txt

```diff
@@ -22,41 +22,76 @@ find_package(Boost REQUIRED
 find_package(PkgConfig)
 pkg_check_modules (MUPARSER REQUIRED muparser)
 
-catkin_package()
+catkin_package(
+  INCLUDE_DIRS
+    include
+  LIBRARIES
+    canopen_motor
+  CATKIN_DEPENDS
+    canopen_402
+    canopen_chain_node
+    controller_manager
+    hardware_interface
+    joint_limits_interface
+    urdf
+  DEPENDS
+    Boost
+    MUPARSER
+)
 
 include_directories(
   include
   ${catkin_INCLUDE_DIRS}
   ${MUPARSER_INCLUDE_DIRS}
 )
 
+# canopen_motor
+add_library(canopen_motor
+  src/controller_manager_layer.cpp
+  src/handle_layer.cpp
+  src/motor_chain.cpp
+  src/robot_layer.cpp
+)
+target_link_libraries(canopen_motor
+  ${catkin_LIBRARIES}
+  ${MUPARSER_LIBRARIES}
+)
+add_dependencies(canopen_motor
+  ${catkin_EXPORTED_TARGETS}
+)
+
 # canopen_motor_node
 add_executable(canopen_motor_node
-  src/control_node.cpp
-  src/robot_layer.cpp
+  src/canopen_motor_chain_node.cpp
 )
 target_link_libraries(canopen_motor_node
+  canopen_motor
   ${catkin_LIBRARIES}
-  ${MUPARSER_LIBRARIES}
```

Comment:
I just realised that canopen_motor_node might need additional lib froms catkin, that have been stripped from canopen_motor.